include/rocksdb/comparator.h

```diff
@@ -33,6 +33,35 @@ class CompareInterface {
   virtual int Compare(const Slice& a, const Slice& b) const = 0;
 };
 
+// The general interface for comparing two Slices that are related to
+// timestamps.
+// They are defined for both Comparator and some internal data structures.
+class CompareWithTimestampInterface {
```

Comment:
timestamp_size should be moved into this class (and the Comparator constructor) should call the appropriate related constructor.

Reply:
This interface may be used a wrapper. If we move timestamp_size there, there will be potential consistency issues.